File: other-langs/stream.js
// simple
let r = [1,2,3,4,5,6,7,8,9].filter(i=>i>5).map(i=>i*10).reduce((a,b)=>a+b, 0)
console.log("Final result: " + r)

// log
r = [1,2,3,4,5,6,7,8,9]
    .filter(i => {
      console.log("Filter: " + i + " > 5 --> " + (i > 5))
      return i > 5
    })
    .map(i => {
      console.log("Map: " + i + " * 10 --> " + (i * 10))
      return i * 10
    })
    .reduce((a, b) => {
      console.log("Reduce: " + a + " + " + b + " --> " + (a + b))
      return a + b
    }, 0)
console.log("Final result: " + r)

File: other-langs/stream.rb
# simple
r = [1,2,3,4,5,6,7,8,9].select {|i| i>5}.map {|i| i*10}.reduce(0) {|a,b| a+b}
puts "Final result: #{r}"

# log
r = [1,2,3,4,5,6,7,8,9]
    .select do |i|
      puts "Filter: #{i} > 5 --> #{i > 5}"
      i > 5
    end
    .map do |i|
      puts "Map: #{i} * 10 --> #{i * 10}"
      i * 10
    end
    .reduce(0) do |a,b|
      puts "Reduce: #{a} + #{b} --> #{a + b}"
      a + b
    end
puts "Final result: #{r}"
